Add Rawhide vs ELN in FAQ

tdawson opened the pull request:
Signed-off-by: Troy Dawson <tdawson@redhat.com>

jkonecny12 commented:
Looks great to me. Thanks!

bookwar commented:
There is a dedicated page which describes the buildroot setup of ELN https://docs.fedoraproject.org/en-US/eln/buildroot/

Maybe we can move the table which describes the buildroot configuration from FAQ there. And link to it from FAQ? So that we have one place for all buildroot information.

sgallagher commented on modules/ROOT/pages/faq.adoc:
```diff
@@ -18,4 +18,36 @@ What is the role of ELN with respect to CentOS Stream?::
+    ELN packages.  They are the same versions, usually built on the same libraries.  But
+    there are some buildroot differences in global macros that can cause build differences.
+    
+    Note: Many packages have %if fedora / rhel conditions that cause Rawhide and ELN packages
+    to differ.  Listing all of those differences is beyond the scope of this document.
```
```suggestion
    Note: Many packages have %if fedora / rhel conditionals in their specfiles that produce differences between the built Rawhide and ELN packages.  Listing all of those differences is beyond the scope of this document.
```

sgallagher commented on modules/ROOT/pages/faq.adoc:
```diff
@@ -18,4 +18,36 @@ What is the role of ELN with respect to CentOS Stream?::
+
+When should I use the rpm macro %\{eln}?::
+
+    Never.  If you think you need to, file a ticket on the 
+    https://github.com/fedora-eln/eln/issues/new/choose[issue tracker].
```
```suggestion
    Never.  In nearly all cases, you actually want to use the %\{rhel} macro. If you think you need to use the %\{eln} macro, file a ticket on the https://github.com/fedora-eln/eln/issues/new/choose[issue tracker] with your justification.
```

tdawson commented:
I used all of Stephens suggestions.
I have moved the buildroot stuff, over to the buildroot page, with links.
I have updated the buildroot page with the latest compiler flag info.

An example is found here:
https://tdawson.fedorapeople.org/eln/public/eln/faq/
https://tdawson.fedorapeople.org/eln/public/eln/buildroot/

sgallagher commented on modules/ROOT/pages/faq.adoc:
```diff
@@ -18,4 +18,22 @@ What is the role of ELN with respect to CentOS Stream?::
 
      
-
+What is the differences between Rawhide and ELN?::
```
```suggestion
What are the differences between Rawhide and ELN?::
```

sgallagher reviewed (changes requested):
Additionally (though this could be a separate change if you prefer), there's a section on the buildroot page that could be updated:

Simply specify `eln` as the target when kicking off a `koji` build using `fedpkg`:
```
fedpkg build [--scratch] --target eln [--srpm [SRPM]] [...]
```

Could be simplified to
Simply specify `eln` as the release when kicking off a `koji` build using `fedpkg`:
```
fedpkg --release eln build [--scratch] [--srpm [SRPM]] [...]
```


I mostly note this because the existing example is actually wrong. You would need to specify both `--target eln` and `--skip-nvr-check` for it to work properly, whereas `--release eln` handles it correctly.

sgallagher reviewed (approved):
LGTM
